assets/javascripts/discourse/components/param-input-form.gjs

```diff
@@ -0,0 +1,322 @@
+
+const layoutMap = {
+  int: "int",
+  bigint: "string",
+  boolean: "boolean",
+  string: "string",
+  time: "generic",
+  date: "generic",
+  datetime: "generic",
+  double: "string",
+  user_id: "user_id",
+  post_id: "string",
+  topic_id: "generic",
+  category_id: "category_id",
+  group_id: "generic",
+  badge_id: "generic",
+  int_list: "generic",
+  string_list: "generic",
+  user_list: "user_list",
+  group_list: "group_list",
+};
+
+const ERRORS = {
+  REQUIRED: I18n.t("form_kit.errors.required"),
+  NOT_AN_INTEGER: I18n.t("form_kit.errors.not_an_integer"),
+  NOT_A_NUMBER: I18n.t("form_kit.errors.not_a_number"),
+  OVERFLOW_HIGH: I18n.t("form_kit.errors.too_high", { count: 2147484647 }),
+  OVERFLOW_LOW: I18n.t("form_kit.errors.too_low", { count: -2147484648 }),
+  INVALID: I18n.t("explorer.form.errors.invalid"),
+  NO_SUCH_CATEGORY: I18n.t("explorer.form.errors.no_such_category"),
+  NO_SUCH_GROUP: I18n.t("explorer.form.errors.no_such_group"),
+};
+
+function digitalizeCategoryId(value) {
+  value = String(value || "");
+  const isPositiveInt = /^\d+$/.test(value);
+  if (!isPositiveInt) {
+    if (/\//.test(value)) {
+      const match = /(.*)\/(.*)/.exec(value);
+      if (!match) {
+        value = null;
+      } else {
+        value = Category.findBySlug(
+          dasherize(match[2]),
+          dasherize(match[1])
+        )?.id;
+      }
+    } else {
+      value = Category.findBySlug(dasherize(value))?.id;
+    }
+  }
+  return value?.toString();
+}
+
+function normalizeValue(info, value) {
+  switch (info.type) {
+    case "category_id":
+      return digitalizeCategoryId(value);
+    case "boolean":
+      if (value == null) {
+        return info.nullable ? "#null" : false;
+      }
+      return value;
+    case "group_list":
+    case "user_list":
+      if (Array.isArray(value)) {
+        return value || null;
+      }
+      return value?.split(",") || null;
+    case "user_id":
+      if (Array.isArray(value)) {
+        return value[0];
+      }
+      return value;
+    default:
+      return value;
+  }
+}
+
+function serializeValue(type, value) {
+  switch (type) {
+    case "string":
+    case "int":
+      return value != null ? String(value) : "";
+    case "boolean":
+      return String(value);
+    case "group_list":
+    case "user_list":
+      return value?.join(",");
+    default:
+      return value?.toString();
+  }
+}
+
+function validationOf(info) {
+  if (info.type === "boolean") {
+    return "";
+  }
+  return info.nullable ? "" : "required";
+}
+
+function componentOf(info) {
+  let type = layoutMap[info.type] || "generic";
+  if (info.nullable && type === "boolean") {
+    type = "boolean_three";
+  }
+  switch (type) {
+    case "int":
+      return <template>
+        <@field.Input @type="number" name={{@info.identifier}} />
+      </template>;
+    case "boolean":
+      return <template><@field.Checkbox name={{@info.identifier}} /></template>;
```

Comment:
That should be a form-kit bug? if this is set to required, you will have no choice but to choose it :eyes: 
![image](https://github.com/user-attachments/assets/a68c6924-2cfc-4181-a69d-d97ee6a03890)

Reply:
I think so too, because we already have [accepted](https://docs-demo.dtaylor.uk/docs/guides/frontend/form-kit/validation#accepted) to indicate a required option.

Reply:
Yes I agree that would be better indeed :+1: